### PR TITLE
Add access to values of properties

### PR DESCRIPTION
**Pull request by alkino**
API can be discussed

### CODE REVIEW

**Comment by alkino** [comment]
I wonder if we should return tuple instead of having one getter for each value.

Should we do a static function instead?

```c++
static std::tuple<hsize_t, hsize_t> fromDataTransferProps(dxpl);
```

File: include/highfive/H5PropertyList.hpp [hunk 1]
```diff
@@ -244,15 +259,29 @@ class FileVersionBounds {
         : _low(low)
         , _high(high) {}
 
+    FileVersionBounds(const FileAccessProps& fapl) {
+        if (H5Pget_libver_bounds(fapl.getId(), &_low, &_high) < 0) {
+            HDF5ErrMapper::ToException<PropertyException>("Unable to access file version bounds");
+        }
+    }
+
+    H5F_libver_t get_low() const {
```
Comment (matz-e) [inline include/highfive/H5PropertyList.hpp:268]: I think here we could produce a `std::pair`

Comment (alkino) [reply include/highfive/H5PropertyList.hpp]: I did ;)

File: include/highfive/bits/H5PropertyList_misc.hpp [hunk 2]
```diff
@@ -394,19 +394,71 @@ inline void AllocationTime::apply(hid_t dcpl) const {
 AllocationTime::AllocationTime(H5D_alloc_time_t alloc_time)
     : _alloc_time(alloc_time) {}
 
+AllocationTime::AllocationTime(const DataSetCreateProps& dcpl) {
+    if (H5Pget_alloc_time(dcpl.getId(), &_alloc_time) < 0) {
+        HDF5ErrMapper::ToException<PropertyException>("Error setting allocation time");
```
[the remark this answers is not in the document]
Comment (1uc) [reply include/highfive/bits/H5PropertyList_misc.hpp:401]: Thank you!

File: include/highfive/bits/H5PropertyList_misc.hpp [hunk 3]
```diff
@@ -189,31 +387,136 @@ inline void Shuffle::apply(const hid_t hid) const {
...
+inline uint32_t MpioNoCollectiveCause::getLocalCause() const {
+    return _local_cause;
+}
+
+inline uint32_t MpioNoCollectiveCause::getGlobalCause() const {
+    return _global_cause;
+}
+
+inline std::pair<uint32_t, uint32_t> MpioNoCollectiveCause::getCause() const {
+    return {_local_cause, _global_cause};
```
Comment (1uc) [inline include/highfive/bits/H5PropertyList_misc.hpp:518]: Isn't this a C++17-feature? If so why is it getting past CI?

**Review by 1uc — APPROVED** [review]

**Pull request merged**